View Controllers/ListViewController.m

```diff
@@ -156,12 +214,49 @@ - (nonnull UITableViewCell *)tableView:(nonnull UITableView *)tableView cellForR
  - (NSInteger)tableView:(nonnull UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
      
      if (self.segmentIndex == 0) {
-         [self setView];
          return self.favorites.count;
      }
      else {
-         [self setView];
          return self.wishlist.count;
      }
- }
+     
+}
+
+- (void)didAddFavorite:(NSNotification*)notification {
+    GMSPlace* place = (GMSPlace*)notification.object;
+    
+    // add to the favorites
+    NSMutableArray<GMSPlace*>* favorites = [NSMutableArray arrayWithArray:self.favorites];
```

Comment:
to make this even easier, you could do self.favorites = [[NSArray arrayWithObject: insertObject:place] arrayByAddingObjectsFromArray:self.favorites].... i didn't double check that syntax but that should work i think :0